src/gluonts/torch/distributions/mqf2.py

```diff
@@ -0,0 +1,395 @@
+        hidden_state
+            hidden_state, shape = (batch_shape, hidden_size)
+
+        Returns
+        -------
+        results
+            predicted paths of shape = (batch_shape, prediction_length)
+        """
+
+        picnn = self.picnn
+        if hidden_state is None:
+            hidden_state = self.hidden_state
+
+        normal_quantile = self.standard_normal.icdf(alpha)
+
+        if self.is_energy_score:
+            result = picnn(normal_quantile, context=hidden_state)
+        else:
+            result = picnn.reverse(normal_quantile, context=hidden_state)
+
+        return result
+
+    @staticmethod
+    def get_numel(tensor_shape: torch.Size) -> int:
+        # Auxiliary function
+        # compute number of elements specified in a torch.Size()
+        return torch.prod(torch.tensor(tensor_shape)).item()
+
+    @property
+    def batch_shape(self) -> torch.Size:
+        # last dimension is the hidden state size
+        return self.hidden_state.shape[:-1]
+
+    @property
+    def event_shape(self) -> Tuple:
+        return ()
+
+    @property
+    def event_dim(self) -> int:
+        return 0
+
+
+class MQF2DistributionOutput(DistributionOutput):
+    distr_cls: type = MQF2Distribution
+
+    @validated()
+    def __init__(
+        self,
+        prediction_length: int,
+        is_energy_score: bool = True,
+        threshold_input: float = 100.0,
+        es_num_samples: int = 50,
+        beta: float = 1.0,
+    ) -> None:
+        super().__init__(self)
+        # A null args_dim to be called by PtArgProj
+        self.args_dim = cast(
+            Dict[str, int],
+            {"null": 1},
+        )
+
+        self.prediction_length = prediction_length
+        self.is_energy_score = is_energy_score
+        self.threshold_input = threshold_input
+        self.es_num_samples = es_num_samples
+        self.beta = beta
+
+    @classmethod
+    def domain_map(
+        cls,
+        hidden_state: torch.Tensor,
+    ) -> Tuple:
+        # A null function to be called by ArgProj
+        return ()
+
+    def distribution(
+        self,
+        picnn: torch.nn.Module,
+        hidden_state: torch.Tensor,
+        loc: Optional[torch.Tensor] = 0,
+        scale: Optional[torch.Tensor] = None,
+    ) -> MQF2Distribution:
+
+        distr = self.distr_cls(
+            picnn,
+            hidden_state,
+            prediction_length=self.prediction_length,
+            threshold_input=self.threshold_input,
+            es_num_samples=self.es_num_samples,
+            is_energy_score=self.is_energy_score,
+            beta=self.beta,
+        )
+
+        if scale is None:
+            return distr
+        else:
+            return TransformedMQF2Distribution(
+                distr, [AffineTransform(loc=loc, scale=scale)]
+            )
+
+    @property
+    def event_shape(self) -> Tuple:
+        return ()
+
+
+class TransformedMQF2Distribution(TransformedDistribution):
```

Comment:
Do the rest of the `TransformedDistribution` functionality actually work here? If no, maybe it is cleaner to NOT inherent from it?

Reply:
We do not use the functionality of `TransformedDistribution`. The inheritance is mainly for using the `__init__` of `TransformedDistribution` to do sanity checks.